server/services/applicationService.ts

```diff
@@ -76,6 +76,35 @@ export default class ApplicationService {
     await client.update(application.id, getApplicationUpdateData(application))
   }
 
+  async appendToList(page: TaskListPage, request: Request) {
+    const errors = page.errors()
+
+    if (Object.keys(errors).length) {
+      throw new ValidationError<typeof page>(errors)
+    } else {
+      const application = await this.findApplication(request.user.token, request.params.id)
+      const client = this.applicationClientFactory(request.user.token)
+
+      const pageName = getPageName(page.constructor)
+      const taskName = getTaskName(page.constructor)
+
+      /* eslint-disable no-underscore-dangle */
+      delete request.body._csrf
+      delete request.body.pageName
+      delete request.body.taskName
+
+      application.data = application.data || {}
+      if (application.data[taskName] && application.data[taskName][pageName]) {
```

Comment:
For people coming fresh to this, do you think it might be clearer if these tests were wrapped in functions with descriptive names?

e.g. I _think_ that `(application.data[taskName] && application.data[taskName][pageName]) ` is asking `isAppendingAnItem()`?